src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java

```diff
@@ -449,9 +482,95 @@ private static String deploy (Request req, Response res) {
                 target);
             logMessageAndHalt(req, HttpStatus.ACCEPTED_202, message);
         }
+
+        // If pelias update is requested, launch pelias update job
+        if (deployment.peliasUpdate) {
+            updatePelias(req, res, job);
+        }
+
         return SparkUtils.formatJobMessage(job.jobId, "Deployment initiating.");
     }
 
+    /**
+     * Start an updatePelias job which will trigger the webhook, then check for status updates.
+     */
+    private static String updatePelias (Request req, Response res, DeployJob deployJob) {
+        // Check parameters supplied in request for validity.
+        Auth0UserProfile userProfile = req.attribute("user");
+        Deployment deployment = getDeploymentWithPermissions(req, res);
+        Project project = Persistence.projects.getById(deployment.projectId);
+        if (project == null) {
+            logMessageAndHalt(req, 400, "Internal reference error. Deployment's project ID is invalid");
+        }
+
+        // Get log upload URI from deploy job
+        AmazonS3URI logUploadS3URI = deployJob.getS3FolderURI();
+
+        // Execute the pelias update job and keep track of it
+        PeliasUpdateJob peliasUpdateJob = new PeliasUpdateJob(userProfile, "Updating Custom Geocoder Database", deployment, logUploadS3URI);
+        JobUtils.heavyExecutor.execute(peliasUpdateJob);
```

Comment:
I think we want to add the pelias job as a child job. That way if the deployment fails for some reason, the OTP data and the Pelias data don't get out of sync? Perhaps it was done this way because the Pelias update takes a long time, but that's maybe a tradeoff that we should discuss.

Reply:
Moved in b17de71f83ac9c47b560d98d89b56e63205537a7, I don't think this causes too much change so should be alright?